arch/xtensa/core/irq_offload.c

```diff
@@ -8,8 +8,6 @@
 #include <zsr.h>
 #include <zephyr/irq.h>
 
-#define CURR_CPU (IS_ENABLED(CONFIG_SMP) ? arch_curr_cpu()->id : 0)
```

Comment:
would be nice to remove it from the tree completely, not only for xtensa, i.e. 

arch/arc/core/irq_offload.c has the same code exactly.

Reply:
Yeah, this dates from the earliest days of SMP before the cpus[] array got unified with the single-core version.